tests/foreman/api/test_gpgkey.py

```diff
@@ -28,3 +41,213 @@ def test_positive_search_in_org(self):
+
+    @tier1
+    @run_only_on('sat')
+    def test_positive_update_content_path(self):
+        """Update GPG key content location to another valid one.
+
+        @Assert: The GPG key content path can be updated.
+
+        @Feature: GPGKey
+        """
+        new_key = get_data_file(VALID_GPG_KEY_BETA_FILE)
+        gpg_key = entities.GPGKey(
+            organization=self.org, content=self.key_path).create()
+        gpg_key.content = new_key
+        gpg_key = gpg_key.update(['content'])
+        self.assertEqual(new_key, gpg_key.content)
+
+    @tier1
+    @run_only_on('sat')
+    def test_positive_update_content_path_to_text(self):
+        """Update GPG key content location to valid content text.
+
+        @Assert: The GPG key content field can be updated.
+
+        @Feature: GPGKey
+        """
+        gpg_key = entities.GPGKey(
+            organization=self.org, content=self.key_path).create()
+        gpg_key.content = self.key_content
+        gpg_key = gpg_key.update(['content'])
+        self.assertEqual(self.key_content, gpg_key.content)
+
+    @tier1
+    @run_only_on('sat')
+    def test_negative_update_name(self):
+        """Attempt to update GPG key name to invalid one
+
+        @Assert: GPG key is not updated
+
+        @Feature: GPGKey
+        """
+        gpg_key = entities.GPGKey(organization=self.org).create()
+        for new_name in invalid_values_list():
+            gpg_key.name = new_name
+            with self.subTest(new_name):
+                with self.assertRaises(HTTPError):
+                    gpg_key.update(['name'])
+
+    @tier1
+    @run_only_on('sat')
+    def test_negative_update_same_name(self):
+        """Attempt to update GPG key name to the name of existing GPG key
+        entity
+
+        @Assert: GPG key is not updated
+
+        @Feature: GPGKey
+        """
+        name = gen_string('alpha')
+        entities.GPGKey(organization=self.org, name=name).create()
+        new_gpg_key = entities.GPGKey(organization=self.org).create()
+        new_gpg_key.name = name
+        with self.assertRaises(HTTPError):
+            new_gpg_key.update(['name'])
+
+    @tier1
+    @run_only_on('sat')
+    def test_negative_update_content(self):
+        """Attempt to update GPG key content to invalid one
+
+        @Assert: GPG key is not updated
+
+        @Feature: GPGKey
+        """
+        gpg_key = entities.GPGKey(
+            organization=self.org, content=self.key_content).create()
+        gpg_key.content = ''
+        with self.assertRaises(HTTPError):
+            gpg_key.update(['content'])
+        self.assertEqual(self.key_content, gpg_key.read().content)
+
+    @tier1
+    @run_only_on('sat')
+    def test_positive_delete(self):
+        """Create a GPG key with different names and then delete it.
+
+        @Assert: The GPG key deleted successfully.
+
+        @Feature: GPGKey
+        """
+        for name in valid_data_list():
+            with self.subTest(name):
```

Comment:
You can drop subtests here as API works with entities by their IDs, no matter what the name was, so we usually don't use subtests for api positive_delete tests (opposed to CLI positive_delete_by_name, where subtests are required)